ci/deploy.sh

```diff
@@ -0,0 +1,107 @@
+#!/bin/bash -e
+#
+# Deploy your branch on VIP Go.
+#
+
+# This script uses various Circle CI and Travis CI environment 
+# variables, Circle prefix their environment variables with 
+# `CIRCLE_` and Travis with `TRAVIS_`. 
+# Documentation:
+# https://circleci.com/docs/1.0/environment-variables/
+# https://docs.travis-ci.com/user/environment-variables/
+
+set -ex
+
+DEPLOY_SUFFIX="-built"
+
+BRANCH="${CIRCLE_BRANCH:-$TRAVIS_BRANCH}"
+SRC_DIR="${TRAVIS_BUILD_DIR:-$PWD}"
+BUILD_DIR="/tmp/vip-go-build"
+
+if [[ -z "$BRANCH" ]]; then
+	echo "No branch specified!"
+	exit 1
+fi
+
+if [[ -d "$BUILD_DIR" ]]; then
+	echo "WARNING: ${BUILD_DIR} already exists. You may have accidentally cached this"
+	echo "directory. This will cause issues with deploying."
+	exit 1
+fi
+
+cd $SRC_DIR
+
+if [[ $CIRCLECI ]]; then
+	CIRCLE_REPO_SLUG="${CIRCLE_PROJECT_USERNAME}/${CIRCLE_PROJECT_REPONAME}";
+fi
+REPO_SLUG=${CIRCLE_REPO_SLUG:-$TRAVIS_REPO_SLUG}
+REPO_SSH_URL="git@github.com:${REPO_SLUG}"
+COMMIT_SHA=${CIRCLE_SHA1:-$TRAVIS_COMMIT}
+DEPLOY_BRANCH="${BRANCH}${DEPLOY_SUFFIX}"
+
+# You can also exclude this on Travis with:
+# if: branch =~ ^.*(?<!-built)$
+# …or on CircleCI with:
+# jobs:
+#  build:
+#    branches:
+#      ignore:
+#        - /^.*(?<!-built)$/
+if [[ "$BRANCH" == *${DEPLOY_SUFFIX} ]]; then
+	echo "WARNING: Attempting to build from branch '${BRANCH}' to deploy '${DEPLOY_BRANCH}', seems like recursion so aborting."
+	exit 0
+fi
+
+echo "Deploying $BRANCH to $DEPLOY_BRANCH"
+
+COMMIT_USER_NAME="$( git log --format=%ce -n 1 $COMMIT_SHA )"
+COMMIT_USER_EMAIL="$( git log --format=%cn -n 1 $COMMIT_SHA )"
+GIT_USER="${DEPLOY_GIT_USER:-$COMMIT_USER_NAME}"
+GIT_EMAIL="${DEPLOY_GIT_EMAIL:-$COMMIT_USER_EMAIL}"
+
+git clone "$REPO_SSH_URL" "$BUILD_DIR"
```

Comment:
A full clone is probably unnecessary (and slow for big repos)? Could likely get away with `init` + `remote add`.